src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -319,6 +270,48 @@ private static ProcessHelper CreateProcHelper(string command, string args, Logge
             return procHelper;
         }
 
+        private static int RunIteration(ProcessHelper setupHelper, ProcessHelper testHelper, ProcessHelper cleanupHelper, Logger logger)
+        {
+            int pid = 0;
+
+            if (setupHelper != null)
+            {
+                logger.Log($"***Iteration Setup***");
+                pid = RunProcess(setupHelper, logger);
+            }
+
+            // no need to run test process if setup failed
+            if (pid != FAILED_PID)
+            {
+                logger.Log($"***Test***");
+                pid = RunProcess(testHelper, logger);
+            }
+
+            // need to clean up despite the result of setup and test
+            if (cleanupHelper != null)
+            {
+                logger.Log($"***Iteration Cleanup***");
+                if(RunProcess(cleanupHelper, logger) == FAILED_PID)
+                {
+                    // cleanup failed
+                    return FAILED_PID;
+                }
+            }
+
+            return pid;
+        }
+
+        private static int RunProcess(ProcessHelper helper, Logger logger)
```

Comment:
I think a more .NET way of doing this would be either to return `(bool, int)` or to return `bool` and have the PID as an out param. Sentinel values like `FAILED_PID` aren't common.

Reply:
Good to know! I'll make it return a tuple.